src/pretix/base/metrics.py

```diff
@@ -0,0 +1,150 @@
+import redis
+
+ns_prefix = "pretix_metrics_"
+
+r = redis.Redis(host=redishost, port=redisport)
+
+
+class MetricsError(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+
+class Metric(object):
+    """
+    Base Metrics Object
+    """
+
+    def __init__(self, name, helpstring, labelnames=[]):
+        self.name = name
+        self.help = helpstring
+        self.labelnames = labelnames
+
+    def __repr__(self):
+        return self.name + "{" + ",".join(self.labelnames) + "}"
+
+    def __str__(self):
+        return self.__repr__()
+
+    def _check_label_consistency(self, labelset):
+        """
+        Checks if the given labelset provides exactly the labels that are required.
+        """
+
+        # test if every required label is provided
+        for labelname in self.labelnames:
+            if labelname not in labelset:
+                raise MetricsError("Label {0} not specified.".format(labelname))
+
+        # now test if no further labels are required
+        if len(labelset) != len(self.labelnames):
+            raise MetricsError("Unknown label used.")
```

Comment:
The error message would be a lot more helpful if it said, which one is unknown. This should be easy if you just use actual ``set`` objects and use some primitive set operation.